x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/support/Hasher.java

```diff
@@ -538,6 +540,8 @@ private static boolean verifyPbkdf2Hash(SecureString data, char[] hash) {
                 .encode(secretKeyFactory.generateSecret(keySpec).getEncoded()));
             final boolean result = CharArrays.constantTimeEquals(computedPwdHash, hashChars);
             return result;
+        } catch (AssertionError ae) {
+            throw new ElasticsearchException("Can't use PBKDF2 implementation from the Security Provider in use.", ae);
```

Comment:
Throw instead of returning false so that this is distinguishable from wrong password, which i believe is desired.

Reply:
Nit: I cannot help trying to make this error message read more like the others. So how about "Error in using PBKDF2 implementation from the chosen Security Provider". Also this string has an extra period at the end while the above one does not :)

Reply:
Sure, why not! I'll change the one below while we're at it

Reply:
I like `Error using PBKDF2 implementation from the selected Security Provider` slightly more if you don't mind :)

Reply:
Sure. My comment on this was really my OCD.